lib/charms/vault_k8s/v0/juju_facade.py

```diff
@@ -315,6 +337,30 @@ def set_secret_expiry(self, expiry: datetime, label: str, id: str | None = None)
         except TransientJujuError:
             raise
 
+    def grant_secret(
+        self,
+        relation: Relation,
+        secret: Secret | None = None,
+        secret_id: str | None = None,
+        secret_label: str | None = None,
+    ) -> None:
+        """Grant read access to the secret to the application related.
+
+        If secret ID or label is provided, it retrieves the secret and grants it to the relation.
+
+        Raises:
+            NoSuchSecretError
+            SecretRemovedError
+        """
+        if secret:
+            secret.grant(relation)
+            return
+        try:
+            secret = self.get_secret(id=secret_id, label=secret_label)
+        except (NoSuchSecretError, SecretRemovedError):
+            raise
```

Comment:
What's the value of such a try/except block?

Reply:
Removed